### PR TITLE
👷 chore: deploy demo app to testflight

### PR DESCRIPTION
**Pull request by jin860315**
`deploy_ios_demo_app.yml`說明：
1. 當有 pr 進 main 或是 release/** 時，會觸發此 action 進行 deploy
2. 先 checkout 到欲執行之分支
3. setup yarn
4. 確認 build-number
5. Replace patch with PR_NUMBER on version name
6. Deploy demo App to TestFlight
7. Send deployed message to Slack
8. 如果有 error Report error to Slack
9. 關於 fastlane 資訊可參考` ios/fastlane/README.md`

有測試過可以 buid demo app 出來

### CODE REVIEW

File: .github/workflows/deploy_ios_demo_app.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+          lfs: true
+      - name: checkoutLFS
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - run: git lfs pull
```
Comment (tp26610) [inline .github/workflows/deploy_ios_demo_app.yml:30]: This repo has no large file, so checkoutLFS is no need
```suggestion

```

File: index.ios.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import 'react-native-gesture-handler';
+import './shim.js';
+import { AppRegistry } from 'react-native';
+import { App } from './src/App';
+
+AppRegistry.registerComponent('CoolWalletAppDemo', () => App);
```
Comment (tp26610) [inline index.ios.js:6]: Is this file necessary?

File: .github/workflows/deploy_ios_demo_app.yml (hunk 1 above)
Comment (jin860315) [reply .github/workflows/deploy_ios_demo_app.yml]: 改了～
844d3b8b3d372faaca09f1b5186f71fcc881e9a8

File: index.ios.js (hunk 2 above)
Comment (jin860315) [reply index.ios.js:6]: 對 如果沒有這個檔案的話 build app 會有 error

**Review by tp26610 — APPROVED** [review]

**Review by a0979470582 — APPROVED** [review]
